src/mson.js

```diff
@@ -0,0 +1,214 @@
+/*
+ * Renders refract elements into MSON.
+ */
+
+import {indent} from './filters';
+
+/*
+ * Get type information for an element, such as the element name, whether
+ * it is required, etc. Returns an array of strings.
+ */
+function getTypeAttributes(element, attributes, parent) {
+  let typeAttributes = [];
+
+  if (element.element !== 'string' && !!parent) {
+    // String is the default type. Any parentless type is probably a
+    // section (# Data Structures or + Attributes) so we don't print
+    // the type there either.
+    typeAttributes.push(element.element);
+  }
+
+  if (attributes) {
+    const tmp = attributes.get('typeAttributes');
+    if (tmp) {
+      typeAttributes = typeAttributes.concat(tmp.toValue() || []);
+    }
+  }
+
+  return typeAttributes;
+}
+
+/*
+ * Handle any element content, e.g. list items, object members, etc.
+ * Note, this recursively calls the `handle` method.
+ */
+function handleContent(element, spaces, marker) {
+  let renderedContent = '';
+  let objectLike = null;
+
+  if (element.content === undefined) {
+    return [renderedContent, objectLike];
+  }
+
+  for (const item of element.content) {
+    // Note: the `initialIndent` option above works because we specifically
+    //       do *not* pass it down the rabbit hole here.
+    if (item.element === 'member') {
+      // This is an object type or something similar.
+      objectLike = true;
+      /* eslint-disable no-use-before-define */
+      renderedContent += handle(item.key.content, item.value, {
+        parent: element,
+        spaces,
+        marker,
+        attributesElement: item,
+      });
+      /* eslint-enable no-use-before-define */
+    } else if (item.element === 'ref') {
+      renderedContent += `${marker} Include ${item.content.href}\n`;
+    } else if (item.element === 'select') {
+      // This is a `OneOf` mutually exclusive type. Currently not
+      // supported as it needs some support upstream in Minim.
+      console.warn('MSON select/option elements are not yet supported!');
+    } else {
+      // This is an array type or something similar.
+      objectLike = false;
+      /* eslint-disable no-use-before-define */
```

Comment:
@pksunkara, the specific problem here is that we essentially have recursion through a call loop:

```js
function a(value) {
  if (value >= 100) { return value; }
  b(value);
}

function b(value) {
  a(value + 1);
}

a(5);
```

In the function `a` it calls an undefined function `b`, but we can't define it at the top via `var b;` because you can't do that **plus** create a new function with the same name (redefining a variable is also a lint error). Because of that, we just shut off the rule for those couple lines. It's really kind of a special edge case.